packages/create-nuxt-app/lib/saofile.js

```diff
@@ -117,14 +117,39 @@ module.exports = {
     })
 
     const generator = this
+    const hasTypeScript = this.answers.language.includes('ts')
+    const hasJest = this.answers.test.includes('jest')
+    const hasAxios = this.answers.features.includes('axios')
     actions.push({
       type: 'modify',
       files: 'package.json',
       handler (data) {
-        return { ...data, ...pkg.load(generator) }
+        const result = { ...data, ...pkg.load(generator) }
+        if (hasTypeScript && hasJest) {
+          result.devDependencies['@types/jest'] = '^26.0.15'
```

Comment:
We can put the dep in pacakge.json of `jest` package dir and use `packages/cna-template/template/frameworks/jest/package.js` to remove it when there is no typescript selected.